### PR TITLE
feat: Implement service behavior according to Client dependency

### LINKED ISSUE #802: Implement service behavior according to Client dependency (introduce protocol)
After implementation https://github.com/hyperledger/aries-framework-go/issues/724 service should behave (act) according to the dependency which it receives through the `e.Continue(c.ProtocolDependency(thID))` function.

### PR DESCRIPTION
**Pull request by soluchok**
* added tests for state machine behavior (`SkipProposal`, `Proposal`, `ProposalUnusual`)
* service logic according to dependency (`ExecuteInbound`,`ExecuteOutbound`)

Closes #802 
Part of #269

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,7 +26,7 @@ GOBIN_PATH=$(abspath .)/build/bin
 MOCKGEN = $(GOBIN_PATH)/mockgen
 
 .PHONY: all
-all: checks generate-openapi-spec unit-test bdd-test
+all: generate-openapi-spec checks unit-test bdd-test
```
Comment (fqutishat) [inline Makefile]: why checks is moved

Comment (fqutishat) [reply Makefile]: just add as first target

Comment (soluchok) [reply Makefile]: Done!

File: pkg/didcomm/protocol/introduce/service.go [hunk 2]
```diff
@@ -95,6 +100,9 @@ type Service struct {
 type Provider interface {
 	OutboundDispatcher() dispatcher.Outbound
 	StorageProvider() storage.Provider
+	// this method should be provided by the introduce client
+	// method should be implemented in didexchange Client
+	SendInvitation(inv *didexchange.Invitation, dest *service.Destination) error
```
Comment (rolsonquadras) [inline pkg/didcomm/protocol/introduce/service.go]: The services are initialized during framework creation by passing the context (eg didExchange service [initialization](https://github.com/hyperledger/aries-framework-go/blob/539fbb4c2c29535d2b531997425aaa2cc95420c1/pkg/framework/aries/default.go#L77)). At that point, I'm not sure we would have this dependency.

**Comment by rolsonquadras** [comment]
@soluchok https://github.com/hyperledger/aries-framework-go/pull/767#issuecomment-556091695 similar comment applies here.

File: pkg/didcomm/protocol/introduce/service.go (hunk 2 above)
Comment (soluchok) [reply pkg/didcomm/protocol/introduce/service.go]: @llorllale @troyronda @rolsonquadras  I am going to get it directly from the context, Are you ok with that?
```
svc, err := ctx.Service(didexchange.DIDExchange)
```

**Pull request closed without merging**